### ISSUE TITLE
cabal check should warn when readme or changelog exist but aren't included in extra-source-files

### ISSUE THREAD

**Issue by sjakobi** [haskell/cabal#3964]
Browsing hackage would be a lot nicer if everyone would upload their READMEs.

Would it be within the scope of `cabal check` to look for a README in the package directory and give a warning/suggestion, if that file isn't listed among the `extra-source-files`?

If such a warning is _not_ within the scope of `cabal check`, is there another command that could be reasonably enhanced with such a warning?

**Comment by ezyang** [comment]
Seems like a reasonable thing to me.

**Comment by 23Skidoo** [comment]
While we're at it, we should also add a warning for missing changelog.

**Comment by ezyang** [comment]
@sjakobi Don't forget to PR your change! :)

**Comment by sjakobi** [comment]
I will! :)

I still need to make my path work with glob patterns in the `extra-source-files` section, but that won't be much work.

**Comment by jneira** [comment]
I think this makes sense @sjakobi do you think your work could be shared as a draft pr? thanks!

**Comment by sjakobi** [comment]
@jneira Apologies for the late response! AFAIK all my work on this issue is in this commit: https://github.com/sjakobi/cabal/commit/7dedf0cc8419c2ad893704828861f1277aa3ede2.

If anyone is interested in completing this work, feel free to use that code in any way you see fit.

**Comment by Mikolaj** [comment]
@sjakobi: thank you very much. I've labelled this correctly, so let's hope somebody picks it up now.

**Comment by Mikolaj** [comment]
I put these in `extra-doc-files`. Is that a bad choice?

**Comment by Mikolaj** [comment]
I think `extra-source-files` is actually a bad choice. Should we warn "are you sure your README/changelog is displayed by your executable or otherwise used when compiling your code?"? I guess too many persons make that mistake, so rather not, especially if we are not ready to scan the code and look for uses of README/changelog (I think this may be done in the future for other functionalities and then we can reconsider). Perhaps there was no `extra-doc-files` initially and only  `extra-source-files`?

**Comment by philderbeast** [comment]
> Browsing hackage would be a lot nicer if everyone would upload their READMEs.

I used to upload readmes but the sometimes their rendering on hackage was not what I wanted so I removed them. I think that is OK if the package `description` is good and there's a link to the `source-repository` or `homepage` where the readme is well rendered.

* https://github.com/haskell/hackage-server/issues/890
* https://github.com/haskell/hackage-server/issues/650
* https://github.com/haskell/hackage-server/issues/565

**Comment by wismill** [comment]
@ulysses4ever A warning for _changelogs_ has been added in #8657. However this issue should maybe be kept opened for further discussion: it would require fine-grained selectionable warnings to implement an exhaustive check.

**Comment by Mikolaj** [comment]
Good idea. Reopening.

**Comment by ulysses4ever** [comment]
Thank you, that works for me!

**Comment by andreasabel** [comment]
@ulysses4ever Do you really want to keep this issue open when its OP has been addressed?  Maybe it would be cleaner to open a new issue with the things you find are lacking still.  That would save one from reading through a lot of now irrelevant comments...

**Comment by ulysses4ever** [comment]
@andreasabel I agree that a new issue with an updated goal would be much more helpful. I don't have energy to act on this at the moment though.

**Comment by ffaf1** [comment]
I can do it.

**Comment by ulysses4ever** [comment]
That would be awesome, @ffaf1 !

**Comment by ffaf1** [comment]
Done!
